src/realm/object-store/sync/impl/sync_metadata.cpp

```diff
@@ -182,7 +183,20 @@ SyncMetadataManager::SyncMetadataManager(std::string path, bool should_encrypt,
         }
     };
 
-    SharedRealm realm = Realm::get_shared_realm(config);
+    m_metadata_config = std::move(config);
+    SharedRealm realm;
+    try {
+        realm = get_realm();
+    }
+    catch (const std::exception& e) {
+        if (!should_encrypt)
+            throw(e);
```

Comment:
Rethrowing exceptions is done with just `throw;` (although it probably shouldn't be needed at all if the correct exception is caught).

Reply:
Using the wrong the encryption fails with RealmFileException. Using a configuration with enrypt = false on a metadata realm that is encrypted also fails with RealmFileException but a different message. So this if/throw intends to show the developer that specific message.

So as of now I have:
```
    catch (const RealmFileException& e) {
        if (!should_encrypt)
            throw;
```